src/battle.ts

```diff
@@ -2472,7 +2473,7 @@ class Battle {
 						else actions += "" + pokeName + " healed its burn!";
 						break;
 					case 'tox':
-						if (poke) poke.statusData.toxicTurns = 0;
+						if (poke) poke.statusData.toxicTurns = null;
```

Comment:
Man is the underlying code here pretty bad though, I'll need to fix it elsewhere.